File: Gestion_Angular_Front_end/src/app/classes/facture-dto.ts
import { FournisseurDto } from 'src/app/classes/fournisseur-dto';
import { ProduitDto } from './produit-dto';
export class FactureDto {
  id: number = 0;
  createdAt: string = '';
  numero: string = '';
  code: string = ' ';
  total: number = 0;
  datecommande: string = '';
  fournisseurDto: FournisseurDto = new FournisseurDto();
  produitDto: ProduitDto = new ProduitDto();
}

File: backend/src/main/java/com/odji/spring_back_end/dto/FactureDto.java
package com.odji.spring_back_end.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class FactureDto {

    private Integer id;
    private String numero;
    private String code;
    private LocalDate datecommande;
    private BigDecimal total;

    // Relation sortante exposée (sans suffixe "Dto")
    private FournisseurDto fournisseur;

    // ❌ Pas de "lignefactures" → endpoint dédié si besoin
}
